src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] values;
```

Comment:
i think here is better to define as Object
```suggestion
    private Object[] values;
```
i think here is better to define as Object, so in constructor we won't need to cast an array.

Reply:
I'm not sure about it, but I'll do it.